sonar/models/mutox/classifier.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# MIT_LICENSE file in the root directory of this source tree.
+
+from dataclasses import dataclass
+from typing import Optional
+
+import torch
+from fairseq2.models.utils.arch_registry import ArchitectureRegistry
+from fairseq2.typing import DataType, Device
+from torch import nn
+
+
+class MutoxClassifier(nn.Module):
+    def __init__(
+        self,
+        model_all,
+    ):
+        super().__init__()
+        self.model_all = model_all
+
+    def forward(self, inputs: torch.Tensor, output_prob: bool = False) -> torch.Tensor:
+        if output_prob:
+            self.model_all.add_module("sigmoid", nn.Sigmoid())
+        else:
+            self.model_all.add_module("linear", nn.Linear(128, 1))
```

Comment:
You are doing something very weird here: changing the model structure during its inference. 
This is incorrect for several reasons:
1. The model needs to have the `Linear(128, 1)` layer regardless of whether we want its output to be a logit or a probability, so you didn't need to remove it from `MutoxClassifierBuilder.build_model`. 
2. When a `Linear(128, 1)` (or any other layer with trainable parameters) is added, its parameters are initialized randomly. When a model is loaded with a `ModelLoader`, it overrides these random parameters with the values loaded from a disk. But if you add the model after this, it stays random. 
3. If you run `forward` several times in a row, the sigmoid or linear layers would be added every time.

So I suggest you do the following:
1. Add the last linear layer back in `MutoxClassifierBuilder.build_model`.
2. In `MutoxClassifier.forward`, you conditionally apply `torch.sigmoid` to the model outputs.